src/main/java/team/gif/gearscout/tba/TbaClient.java

```diff
@@ -0,0 +1,222 @@
+package team.gif.gearscout.tba;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.json.*;
+
+import javax.net.ssl.*;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.X509Certificate;
+import java.net.http.*;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpClient.Redirect;
+import java.net.URISyntaxException;
+import java.time.Duration;
+import java.io.StringReader;
+import java.util.HashMap;
+import java.util.ArrayList;
+import java.util.Optional;
+
+public class TbaClient {
+    private static final Logger logger = LogManager.getLogger(TbaClient.class);
+    private static final String baseUri = "https://www.thebluealliance.com/api/v3/";
+    private static final long minCacheAgeSeconds = 1800; // we don't need match results, only the schedule, so a slow refresh is fine
+
+    private final HttpClient httpClient = getHttpClient();
+
+	private String tbaApiKey;
+    private HashMap<String, GetAllMatchesForEventResponse> cachedMatchResponses;
+
+    public TbaClient(String tbaApiKey) {
+        this.tbaApiKey = tbaApiKey;
+        this.cachedMatchResponses = new HashMap();
```

Comment:
A bit pedantic, but you should specify an empty diamond here `new HashMap<>()` to prevent a compiler warning. It has to do with legacy support for pre-generics classes:
https://stackoverflow.com/questions/4166966/what-is-the-point-of-the-diamond-operator-in-java